editor/editor_properties_array_dict.cpp

```diff
@@ -517,6 +543,59 @@ void EditorPropertyArray::drop_data_fw(const Point2 &p_point, const Variant &p_d
 		emit_changed(get_edited_property(), array, "", false);
 		update_property();
 	}
+
+	if (drag_data.has("type") && String(drag_data["type"]) == "nodes") {
+		Vector<Variant> nodes = drag_data["nodes"];
+		Variant node_paths = object->get_array();
+		Node *base_node = get_base_node();
+
+		if (!node_paths.is_array()) {
+			initialize_array(node_paths);
+		} else {
+			node_paths = node_paths.duplicate();
+		}
+
+		for (int i = 0; i < nodes.size(); i++) {
+			Node *node = get_node(nodes[i]);
+			NodePath path = node->get_path();
+
+			if (subtype == Variant::OBJECT) {
+				if (!base_node) {
+					node_paths.call("push_back", get_tree()->get_edited_scene_root()->get_node(path));
+				} else {
+					node_paths.call("push_back", base_node->get_node(path));
+				}
+			} else if (subtype == Variant::NODE_PATH) {
+				if (!base_node) {
+					node_paths.call("push_back", get_tree()->get_edited_scene_root()->get_path_to(node));
+				} else {
+					node_paths.call("push_back", base_node->get_path_to(node));
+				}
+			}
+		}
+
+		emit_changed(get_edited_property(), node_paths, "", false);
+		update_property();
+	}
+}
+
+Node *EditorPropertyArray::get_base_node() {
+	Node *base_node = Object::cast_to<Node>(get_edited_object());
+
+	if (!base_node) {
+		base_node = Object::cast_to<Node>(InspectorDock::get_inspector_singleton()->get_edited_object());
+	}
+	if (!base_node) {
+		// Try a base node within history.
+		if (EditorNode::get_singleton()->get_editor_selection_history()->get_path_size() > 0) {
+			Object *base = ObjectDB::get_instance(EditorNode::get_singleton()->get_editor_selection_history()->get_path_object(0));
+			if (base) {
+				base_node = Object::cast_to<Node>(base);
+			}
+		}
+	}
```

Comment:
Not sure if this is a good idea. It might fetch a random unrelated node.

Reply:
This is coming from and already in the `get_base_node()` method from `EditorPropertyNodePath` inside of `editor_properties.cpp` ...

Reply:
I'm wondering if we should not just call this function directly from one of the `EditorPropertyNodePath` as it will result in different behaviour (`get_edited_object` is different) or at least it should be adapted to remove the first try with `get_edited_object` that will fail for `EditorPropertyNodePath` as it will find a `EditorPropertyArrayObject` which is not a node. 

For the last part it effectively seems dubious at best but since it is already there let's keep it for now.